### PR TITLE
Move all JS in SDK files to TS

### PR DESCRIPTION
**Pull request by infomiho**
- ./core/stitches.config.js (produced an error on import)
    - Rename helped ✅
- ./core/auth.js (produced an error on import)
    - Rename helped ✅
- ./core/config.js (produced an error on import)
    - Rename helped ✅
- ./core/HttpError.js(produced an error on import)
    - Rename + some changes helped ✅
- ./core/AuthError.js (produced an error on import)
    - Rename + some changes helped ✅
- ./universal/validators.js (produced an error on import)
    - Rename + typing helped ✅
- ./server/config.js (produced an error on import)
    - Rename + typing helped ✅
- ./operations/updateHandlersMap.js (produced an error on import)
    - Did not migrate since it's an internal module and to make us faster 🔴
- ./operations/resources.js (produced an error on import)
    - Did not migrate since it's an internal module and to make us faster 🔴
- ~~./rpc/queries/core.js~~ (already has d.ts file)
- ~~./rpc/actions/core.js~~  (already has d.ts file)

### CODE REVIEW

File: waspc/data/Generator/templates/sdk/auth/utils.ts [hunk 1]
```diff
@@ -180,7 +180,7 @@ export async function doFakeWork(): Promise<unknown> {
 
 export function rethrowPossibleAuthError(e: unknown): void {
   if (e instanceof AuthError) {
-    throwValidationError((e as any).message);
```
Comment (infomiho) [inline waspc/data/Generator/templates/sdk/auth/utils.ts:183]: Now properly typed, no need for this 👍

File: waspc/data/Generator/templates/sdk/operations/resources.ts [hunk 2]
```diff
@@ -5,7 +5,7 @@ import { hashQueryKey } from '@tanstack/react-query'
 // Map where key is resource name and value is Set
```
Comment (infomiho) [inline waspc/data/Generator/templates/sdk/operations/resources.ts:5]: @sodic This needs your input

File: waspc/data/Generator/templates/sdk/operations/updateHandlersMap.ts [hunk 3]
```diff
@@ -1,22 +1,33 @@
-export function makeUpdateHandlersMap(calculateHash) {
```
Comment (infomiho) [inline waspc/data/Generator/templates/sdk/operations/updateHandlersMap.ts:1]: @sodic This file needs your input

File: waspc/data/Generator/templates/sdk/operations/resources.ts (hunk 2 above)
Comment (infomiho) [reply waspc/data/Generator/templates/sdk/operations/resources.ts:5]: This is not user facing API, I can revert this change if you like

File: waspc/data/Generator/templates/sdk/operations/updateHandlersMap.ts (hunk 3 above)
Comment (infomiho) [reply waspc/data/Generator/templates/sdk/operations/updateHandlersMap.ts:1]: This is not user facing API, I can revert this change if you like

**Review by Martinsos — COMMENTED** [review]
Gave it a very quick glance but looks good to me!

File: waspc/data/Generator/templates/sdk/server/config.ts [hunk 4]
```diff
@@ -28,10 +53,10 @@ const config = {
   production: getProductionConfig(),
 }
 
-const resolvedConfig = merge(config.all, config[env])
+const resolvedConfig = merge(config.all, config[env]) as Config
```
Comment (infomiho) [inline waspc/data/Generator/templates/sdk/server/config.ts:56]: This was the simplest option

**Pull request merged**